LanguageFeatures/Augmentation-libraries/augmenting_constructors_A02_t01_lib.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if:
+/// ...
+/// - The augmenting constructor parameters specify any default values.
+///
+/// @description Checks that it is a compile-time error if the augmenting
+/// constructor parameters specify any default values.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmenting_constructors_A02_t01.dart';
+
+augment class C {
+  augment C([int x = 0]);
+//               ^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+  augment C.c1({int x = 0});
+//                  ^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+  augment factory C.c2([int x = 0]);
```

Comment:
The problem would be that we get a syntax error and then it would look like we got the static semantic error which is "can't augment a non-factory constructor with a factory" (that is, the test succeeds for the wrong reason).